docs/upgrading.md

```diff
@@ -6,6 +6,72 @@ the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summar
 
 ## Upgrading to v3.3
 
+### feat!: Remove deprecated config flags
+
+This PR removes the following configmap items -
+
+- executorImage (use executor.image in configmap instead)
+  e.g.
+  Workflow controller configmap similar to the following one given below won't be valid anymore:
+
+  ```yaml
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: workflow-controller-configmap
+  data:
+    ...
+    executorImage: argoproj/argocli:latest
+    ...
+  ```
+
+  From now and onwards, only provide the executor image in workflow controller as a command argument as shown below:
+
+  ```yaml
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: workflow-controller-configmap
+  data:
+    ...
+    executor: |
+      image: argoproj/argocli:latest
+    ...
+  ```
+
+- executorImagePullPolicy (use executor.imagePullPolicy in configmap instead)
+  e.g.
+  Workflow controller configmap similar to the following one given below won't be valid anymore:
+
+  ```yaml
+  data:
+    ...
+    executorImagePullPolicy: IfNotPresent
+    ...
+  ```
+
+  Change it as shown below:
+
+  ```yaml
+  data:
+    ...
+    executor: |
+      imagePullPolicy: IfNotPresent
+    ...
+  ```
+
+- executorResources (use executor.resources in configmap instead)
+  It's similar to executorImage and executorImagePullPolicy.
+- namespace
+  E.g. Workflow controller configmap similar to the one given below won't be valid anymore:
```

Comment:
is this correct?

Reply:
Are you talking about executorResources or namespace?

Reply:
Ah. I did not realise than `namespace` was marked as DEPRECATED. I think all settings should be editable via the configmap. So I think, rather than remove namespace, we should leave it, i.e un-deprecated it.

Reply:
No issues.